### PR TITLE
Processing Data in Model

### PR DESCRIPTION
**Pull request by leonardoeng13**
## Description
Now it's time to set up our models. To start with, we have tables for Users, Posts, Comments, and Likes, which means we need a model for each one. You've already set the foreign key in the table schema but be sure to write it here too!

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,5 @@
 class ApplicationController < ActionController::Base
+  def current_user
+    User.take
```
Comment (dasileker) [inline app/controllers/application_controller.rb]: - [ ] The application record has a relation to the migration, I don't know what the **user.take** will do but it certainly does not render the user in your database.

File: app/models/post.rb [hunk 2]
```diff
@@ -1,3 +1,19 @@
...
+  has_many :comments, foreign_key: 'post_id', class_name: 'Comment'
+  has_many :likes, foreign_key: 'post_id', class_name: 'Like'
+
+  def update_comments_counter
+    comments_counter = Comment.where(post_id: id).count
+    update(comments_counter: comments_counter)
+  end
+
+  def update_likes_counter
+    likes_counter = Like.where(post_id: id).count
```
Comment (dasileker) [inline app/models/post.rb]: - [ ] you don't need the line 12 at all.

File: app/models/user.rb [hunk 3]
```diff
@@ -1,2 +1,15 @@
...
+  has_many :likes, foreign_key: 'author_id', class_name: 'Like'
+
+  def most_recent_posts
+    posts = Post.all
+    posts.order(created_at: :desc).limit(3)
+  end
+
+  def update_posts_counter
+    posts_counter = Post.where(author_id: id).count
+    update(posts_counter: posts_counter)
```
Comment (dasileker) [inline app/models/user.rb]: - [ ] You can use the same as the post.rb and the like.rb posts.increment......

File: app/models/user.rb [hunk 4]
```diff
@@ -1,2 +1,15 @@
 class User < ApplicationRecord
+  has_many :posts
+  has_many :comments, foreign_key: 'author_id', class_name: 'Comment'
+  has_many :likes, foreign_key: 'author_id', class_name: 'Like'
+
+  def most_recent_posts
+    posts = Post.all
```
Comment (dasileker) [inline app/models/user.rb:7]: - [ ] Also again you don't need this Line.

File: app/models/user.rb [hunk 5]
```diff
@@ -1,2 +1,15 @@
 class User < ApplicationRecord
+  has_many :posts
+  has_many :comments, foreign_key: 'author_id', class_name: 'Comment'
+  has_many :likes, foreign_key: 'author_id', class_name: 'Like'
+
```
Comment (dasileker) [inline app/models/user.rb]: - [ ] Since the user has the foreign_key's no need for the class_mane's.

File: db/seeds.rb [hunk 6]
```diff
@@ -6,3 +6,22 @@
...
+fourth_post = Post.create(user_id: second_user, title: 'Ola 2', text: 'Este é o segundo post.')
+fifth_post = Post.create(user_id: third_user, title: 'Hallo', text: 'Este é o segundo post.')
+
+Comment.create(post: first_post, user_id: second_user, text: 'Hi Leo!' )
+Comment.create(post: first_post, user_id: third_user, text: 'Hi Leo!' )
+Comment.create(post: second_post, user_id: third_user, text: 'Hi Leo!' )
+Comment.create(post: third_post, user_id: first_user, text: 'Hi Tom!' )
+Comment.create(post: third_post, user_id: third_user, text: 'Hi Tom!' )
+Comment.create(post: fifth_post, user_id: first_user, text: 'Hi Lilly!' )
+Comment.create(post: fifth_post, user_id: second_user, text: 'Hi Lilly!' )
```
Comment (dasileker) [inline db/seeds.rb]: - [ ] The **user_id** is not in any of your databases so I suggest making sure the seed takes the right value for the users. create

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi  @leonardoeng13,** 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N./A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AshakaE — APPROVED** [review]
### Hi @leonardoeng13   🙋‍♀️, 

# STATUS APPROVED ✔

Thanks for making the changes 👏 
   
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**